cmake/pico_pre_load_platform.cmake

```diff
@@ -15,22 +15,28 @@ set(PICO_SAVED_PLATFORM "${PICO_PLATFORM}")
 
 # If PICO_PLATFORM is specified but not PICO_BOARD, we'll make a stab at defaulting
 if (NOT PICO_DEFAULT_BOARD_rp2040)
+    # PICO_CMAKE_CONFIG: PICO_DEFAULT_BOARD_rp2040, The default PICO_BOARD for PICO_PLATFORM==rp2040, type=string, default=pico, group=build, docref=cmake-platform-board-config
     set(PICO_DEFAULT_BOARD_rp2040 "pico")
 endif()
 if (NOT PICO_DEFAULT_BOARD_rp2350)
+    # PICO_CMAKE_CONFIG: PICO_DEFAULT_BOARD_rp2350, The default PICO_BOARD for PICO_PLATFORM==rp2350, type=string, default=pico2, group=build, docref=cmake-platform-board-config
     set(PICO_DEFAULT_BOARD_rp2350 "pico2")
 endif()
 if (NOT PICO_DEFAULT_BOARD_rp2350-arm-s)
+    # PICO_CMAKE_CONFIG: PICO_DEFAULT_BOARD_rp2350-arm-s, The default PICO_BOARD for PICO_PLATFORM==rp2350-arm-s, type=string, default=pico2, group=build, docref=cmake-platform-board-config
     set(PICO_DEFAULT_BOARD_rp2350-arm-s "pico2")
 endif()
 if (NOT PICO_DEFAULT_BOARD_rp2350-riscv)
+    # PICO_CMAKE_CONFIG: PICO_DEFAULT_BOARD_rp2350-riscv, The default PICO_BOARD for PICO_PLATFORM==rp2350-riscv, type=string, default=pico2, group=build, docref=cmake-platform-board-config
     set(PICO_DEFAULT_BOARD_rp2350-riscv "pico2")
 endif()
 if (NOT PICO_DEFAULT_BOARD_host)
+    # PICO_CMAKE_CONFIG: PICO_DEFAULT_BOARD_host, The default PICO_BOARD for PICO_PLATFORM==host, type=string, default=none, group=build, docref=cmake-platform-board-config
     set(PICO_DEFAULT_BOARD_host "none")
 endif()
 
 if (NOT PICO_DEFAULT_PLATFORM)
```

Comment:
Again, perhaps it's not worth setting `PICO_DEFAULT_PLATFORM` if an explicit `PICO_PLATFORM` has been set?